addons/spectator/functions/fnc_onDisplayLoad.sqf

```diff
@@ -12,19 +12,51 @@
  * Public: No
  */
 
+#define COUNTER_TEXT_DISABLED "  --:--.---"
+
 params [["_display", displayNull]];
 
 // prevent gametime and respawn delay overlap
 private _counterDisplay = uiNamespace getVariable ["RscRespawnCounter", displayNull];
 if (!isNull _counterDisplay) then {
+
+    private _counterTxtCtrl = _counterDisplay displayCtrl IDC_COUNTER_TXT;
     private _counterBgCtrl = _counterDisplay displayCtrl IDC_COUNTER_BG;
+    private _counterBg2Ctrl = _counterDisplay displayCtrl IDC_COUNTER_BG2;
+
+    if (ctrlText _counterTxtCtrl == COUNTER_TEXT_DISABLED) exitWith {
```

Comment:
> Respawn counter will not show when respawn is disabled
It still does show if respawn is disabled when spectator interface is up but I don't think it's a big problem.

Reply:
Yes, I decided to ignore that case as it's really rare.